x-pack/plugins/discover_log_explorer/public/customizations/custom_dataset_selector.tsx

```diff
@@ -5,28 +5,37 @@
  * 2.0.
  */
 
-import React from 'react';
+/* eslint-disable react-hooks/exhaustive-deps */
+
+import React, { useState } from 'react';
 import { DiscoverStateContainer } from '@kbn/discover-plugin/public';
-import { IndexPattern } from '@kbn/io-ts-utils';
-import { Dataset } from '../../common/datasets/models/dataset';
-import { DatasetSelectionHandler, DatasetSelector } from '../components/dataset_selector';
+import { DatasetSelector } from '../components/dataset_selector';
 import { DatasetsProvider, useDatasetsContext } from '../hooks/use_datasets';
-import { InternalStateProvider, useDataView } from '../hooks/use_data_view';
+import { InternalStateProvider } from '../hooks/use_data_view';
 import { IntegrationsProvider, useIntegrationsContext } from '../hooks/use_integrations';
 import { IDatasetsClient } from '../services/datasets';
+import {
+  AllDatasetSelection,
+  DatasetSelection,
+  DatasetSelectionChange,
+} from '../utils/dataset_selection';
 
 interface CustomDatasetSelectorProps {
   stateContainer: DiscoverStateContainer;
 }
 
 export const CustomDatasetSelector = withProviders(({ stateContainer }) => {
-  // Container component, here goes all the state management and custom logic usage to keep the DatasetSelector presentational.
-  const dataView = useDataView();
+  /**
+   * TOREMOVE: This is a temporary workaround to control the datasetSelection value
+   * until we handle the restore/initialization of the dataview with https://github.com/elastic/kibana/issues/160425,
+   * where this value will be used to control the DatasetSelector selection with a top level state machine.
+   */
+  const [datasetSelection, setDatasetSelection] = useState<DatasetSelection>(() =>
```

Comment:
**note:** as mentioned above, this state is only used to control the selection state, in the next task we'll move this logic and its initialization to a state machine that restores from the URL instead of hardcoding the AllDatasetSelection selection type.